src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -355,8 +355,14 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             {
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.info);
             }
+
             info.AddValue(VersionName, version);
-            info.AddValue(ComparerName, comparer, typeof(IEqualityComparer<TKey>));
+            // NonRandomizedStringEqualityComparer is not deserializable in Desktop,
```

Comment:
Do we want to make NonRandomizedStringEqualityComparer non-serializable in .NET Core as well?

Reply:
Yes, just checked Desktop sources. It's really not there, so it should be removed.

Reply:
Or if you want to keep NonRandomizedStringEqualityComparer serializable, it may be better to do this change in NonRandomizedStringEqualityComparer GetObjectData so that it is same as what's done for all other comparers:

In any case, you should be able to remove TypeForwardedFrom attribute on NonRandomizedStringEqualityComparer because of it is not necessary there anymore.

Reply:
> Yes, just checked Desktop sources. It's really not there, so it should be removed.

Depends whether you want to maintain backward compatibility with this bug - allow blob serialized on .NET Core 2.0.0 to be deserialized on .NET Core 2.1,..

Reply:
Though decision. I think closing the bug is to impactful as Dictionary with string as a key is pretty common... I will remove the TypeForwardedFrom attribute and make the change in NRSEC as you suggested. That's a great idea :) Thanks